sys/net/application_layer/cord/ep/cord_ep.c

```diff
@@ -91,7 +90,7 @@ static void _on_register(const gcoap_request_memo_t *memo, coap_pkt_t* pdu,
         /* read the location header and save the RD details on success */
         if (coap_get_location_path(pdu, (uint8_t *)_rd_loc,
                                    sizeof(_rd_loc)) > 0) {
-            memcpy(&_rd_remote, remote, sizeof(_rd_remote));
+            memcpy(&_rd_remote, pdu->remote, sizeof(_rd_remote));
```

Comment:
Maybe it makes sense to have a dedicated function for this. This function could `assert()` that the endpoint indeed of a UDP flavor, once multiple protocols are supported.